### ISSUE TITLE
F0002 crash when parsing a specific file

### ISSUE THREAD

**Issue by romanvm** [PyCQA/pylint#7917]
### Bug description

Pylint crashes when parsing this file: https://github.com/romanvm/kodi.tvmaze/blob/master/metadata.tvmaze/libs/exception_logger.py

Here's the link to the crash report: https://github.com/romanvm/kodi.tvmaze/actions/runs/3663603645/jobs/6193488753#step:5:25

### Command used

```shell
pylint metadata.tvmaze/libs metadata.tvmaze/main.py
```

### Pylint output

```shell
Exception on node <For l.47 at 0x7fc562428160> in file '/home/runner/work/kodi.tvmaze/kodi.tvmaze/metadata.tvmaze/libs/exception_logger.py'
Traceback (most recent call last):
  File "/opt/hostedtoolcache/Python/3.9.15/x64/lib/python3.9/site-packages/pylint/utils/ast_walker.py", line 90, in walk
    callback(astroid)
  File "/opt/hostedtoolcache/Python/3.9.15/x64/lib/python3.9/site-packages/pylint/checkers/refactoring/refactoring_checker.py", line 682, in visit_for
    self._check_unnecessary_list_index_lookup(node)
  File "/opt/hostedtoolcache/Python/3.9.15/x64/lib/python3.9/site-packages/pylint/checkers/refactoring/refactoring_checker.py", line 2118, in _check_unnecessary_list_index_lookup
    has_start_arg, confidence = self._enumerate_with_start(node)
  File "/opt/hostedtoolcache/Python/3.9.15/x64/lib/python3.9/site-packages/pylint/checkers/refactoring/refactoring_checker.py", line 2236, in _enumerate_with_start
    start_val, confidence = self._get_start_value(start_arg)
  File "/opt/hostedtoolcache/Python/3.9.15/x64/lib/python3.9/site-packages/pylint/checkers/refactoring/refactoring_checker.py", line 2260, in _get_start_value
    start_val = node.value
AttributeError: 'BinOp' object has no attribute 'value'
************* Module /home/runner/work/kodi.tvmaze/kodi.tvmaze/.pylintrc
.pylintrc:1:0: E0015: Unrecognized option found: no-space-check (unrecognized-option)
************* Module libs.data_service
metadata.tvmaze/libs/data_service.py:438:0: R1710: Either all return statements in a function should return an expression, or none of them should. (inconsistent-return-statements)
Can't write the issue template for the crash in /home/runner/.cache/pylint/pylint-crash-2022-12-10-09-26-36.txt because of: '[Errno 2] No such file or directory: '/home/runner/.cache/pylint/pylint-crash-2022-12-10-09-26-36.txt''
Here's the content anyway:
First, please verify that the bug is not already filled:
https://github.com/PyCQA/pylint/issues/

Then create a new crash issue:
https://github.com/PyCQA/pylint/issues/new?assignees=&labels=crash%2Cneeds+triage&template=BUG-REPORT.yml


Issue title:
Crash ` (if possible, be more specific about what made pylint crash)
Content:
When parsing the following file:

<!--
 If sharing the code is not an option, please state so,
 but providing only the stacktrace would still be helpful.
 -->


# (c) Roman Miroshnychenko <roman1972@gmail.com> 2020
#
# This program is free software: you can redistribute it and/or modify
# it under the terms of the GNU General Public License as published by
# the Free Software Foundation, either version 3 of the License, or
# (at your option) any later version.
    walker.walk(node)
  File "/opt/hostedtoolcache/Python/3.9.15/x64/lib/python3.9/site-packages/pylint/utils/ast_walker.py", line 93, in walk
    self.walk(child)
  File "/opt/hostedtoolcache/Python/3.9.15/x64/lib/python3.9/site-packages/pylint/utils/ast_walker.py", line 93, in walk
    self.walk(child)
  File "/opt/hostedtoolcache/Python/3.9.15/x64/lib/python3.9/site-packages/pylint/utils/ast_walker.py", line 93, in walk
    self.walk(child)
  File "/opt/hostedtoolcache/Python/3.9.15/x64/lib/python3.9/site-packages/pylint/utils/ast_walker.py", line 90, in walk
    callback(astroid)
  File "/opt/hostedtoolcache/Python/3.9.15/x64/lib/python3.9/site-packages/pylint/checkers/refactoring/refactoring_checker.py", line 682, in visit_for
    self._check_unnecessary_list_index_lookup(node)
  File "/opt/hostedtoolcache/Python/3.9.15/x64/lib/python3.9/site-packages/pylint/checkers/refactoring/refactoring_checker.py", line 2118, in _check_unnecessary_list_index_lookup
    has_start_arg, confidence = self._enumerate_with_start(node)
  File "/opt/hostedtoolcache/Python/3.9.15/x64/lib/python3.9/site-packages/pylint/checkers/refactoring/refactoring_checker.py", line 2236, in _enumerate_with_start
    start_val, confidence = self._get_start_value(start_arg)
  File "/opt/hostedtoolcache/Python/3.9.15/x64/lib/python3.9/site-packages/pylint/checkers/refactoring/refactoring_checker.py", line 2260, in _get_start_value
    start_val = node.value
AttributeError: 'BinOp' object has no attribute 'value'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/opt/hostedtoolcache/Python/3.9.15/x64/lib/python3.9/site-packages/pylint/lint/pylinter.py", line 755, in _lint_files
    self._lint_file(fileitem, module, check_astroid_module)
  File "/opt/hostedtoolcache/Python/3.9.15/x64/lib/python3.9/site-packages/pylint/lint/pylinter.py", line 792, in _lint_file
    raise astroid.AstroidError from e
astroid.exceptions.AstroidError
```
.
************* Module libs.exception_logger
metadata.tvmaze/libs/exception_logger.py:1:0: F0002: metadata.tvmaze/libs/exception_logger.py: Fatal error while checking 'metadata.tvmaze/libs/exception_logger.py'. Please open an issue in our bug tracker so we address this. There is a pre-filled template that you can use in '/home/runner/.cache/pylint/pylint-crash-2022-12-10-09-26-36.txt'. (astroid-error)
************* Module libs.cache_service
metadata.tvmaze/libs/cache_service.py:18:0: C0413: Import "import json" should be placed at the top of the module (wrong-import-position)
metadata.tvmaze/libs/cache_service.py:19:0: C0413: Import "import os" should be placed at the top of the module (wrong-import-position)
metadata.tvmaze/libs/cache_service.py:20:0: C0413: Import "import time" should be placed at the top of the module (wrong-import-position)
metadata.tvmaze/libs/cache_service.py:21:0: C0413: Import "from typing import Optional, Text, Dict, Any, Union" should be placed at the top of the module (wrong-import-position)
metadata.tvmaze/libs/cache_service.py:23:0: C0413: Import "import xbmcgui" should be placed at the top of the module (wrong-import-position)
metadata.tvmaze/libs/cache_service.py:24:0: C0413: Import "import xbmcvfs" should be placed at the top of the module (wrong-import-position)
metadata.tvmaze/libs/cache_service.py:31:0: C0413: Import "from .utils import ADDON_ID, logger" should be placed at the top of the module (wrong-import-position)
metadata.tvmaze/libs/cache_service.py:18:0: C0411: standard import "import json" should be placed before "import xbmc" (wrong-import-order)
metadata.tvmaze/libs/cache_service.py:19:0: C0411: standard import "import os" should be placed before "import xbmc" (wrong-import-order)
metadata.tvmaze/libs/cache_service.py:20:0: C0411: standard import "import time" should be placed before "import xbmc" (wrong-import-order)
metadata.tvmaze/libs/cache_service.py:21:0: C0411: standard import "from typing import Optional, Text, Dict, Any, Union" should be placed before "import xbmc" (wrong-import-order)
metadata.tvmaze/libs/cache_service.py:23:0: C0411: third party import "import xbmcgui" should be placed before "from .utils import logger" (wrong-import-order)
metadata.tvmaze/libs/cache_service.py:24:0: C0411: third party import "import xbmcvfs" should be placed before "from .utils import logger" (wrong-import-order)
metadata.tvmaze/libs/cache_service.py:21:0: C0412: Imports from package typing are not grouped (ungrouped-imports)
```


### Expected behavior

No crash

### Pylint version

```shell
Pylint 2.15.8

Python 3.9.15
```

### OS / Environment

Github Actions Python 3.9 runner

**Comment by DanielNoord** [comment]
@clavedeluna I have lost track. Did we already fix this issue?

**Comment by Pierre-Sassoulas** [comment]
I can reproduce on 2.15.8 and I cannot on main. This will be resolved in pylint 2.16.0.
